File: Juego/Invasiones/fuente/GUI/Aro.cs
using System;
using System.Collections.Generic;
using System.Text;
using Invasiones.Nivel;
using Invasiones.Sprites;
using Invasiones.Dibujo;
using System.Drawing;

namespace Invasiones.GUI
{
    public class Aro : Objeto
    {
        private Animaciones m_animacion;

        public Aro (Animaciones anim, int i, int j)
        {


            m_posEnTileFisico.X = i;
            m_posEnTileFisico.Y = j;

            Point p = TransformarIJEnXY(m_posEnTileFisico.X, m_posEnTileFisico.Y);

            m_posEnMundoPlano.X = p.X;
            m_posEnMundoPlano.Y = p.Y;

          

            m_animacion = anim;
            m_animacion.Cargar();
            ActualizarPosicionXY();
            m_posEnMundoPlano.X -= m_animacion.Offsets.X;
            m_posEnMundoPlano.Y -= m_animacion.Offsets.Y;
           
            m_animacion.Reproducir();
            m_animacion.Loop = true;
        }

        public override void Actualizar()
        {
            base.Actualizar();

            m_animacion.Actualizar();
        }

        
        public override void  Dibujar(Video g)
        {
            if (m_animacion != null)
            {
                m_animacion.Dibujar(g, m_x + s_mapa.TileAncho / 2, m_y + s_mapa.TileAlto / 2, 0);
            }
        }

    
        public void SetearPosicion(int i, int j)
        {
            m_posEnTileFisico.X = i;
            m_posEnTileFisico.Y = j;

            Point p = TransformarIJEnXY(m_posEnTileFisico.X, m_posEnTileFisico.Y);

            m_posEnMundoPlano.X = p.X;
            m_posEnMundoPlano.Y = p.Y;

            m_posEnMundoPlano.X -= m_animacion.Offsets.X ;
            m_posEnMundoPlano.Y -= m_animacion.Offsets.Y;

            ActualizarPosicionXY();
        }
    }
}

File: macOS/Invasiones/Invasiones/GUI/Aro.swift
// GUI/Aro.swift
// Puerto de Aro.cs — aro animado que indica el objetivo actual en el mapa.

import Foundation

class Aro: Objeto {

    // MARK: - Declaraciones
    private let m_animacion: Animaciones

    // MARK: - Constructor
    init(_ anim: Animaciones, _ i: Int, _ j: Int) {
        m_animacion = anim
        super.init()

        m_posEnTileFisico = (i, j)
        let p = transformarIJEnXY(i, j)
        m_posEnMundoPlano = p

        m_animacion.cargar()
        actualizarPosicionXY()

        m_posEnMundoPlano.x -= m_animacion.offsets.x
        m_posEnMundoPlano.y -= m_animacion.offsets.y

        m_animacion.reproducir()
        m_animacion.loop = true
    }

    // MARK: - Override

    @discardableResult
    override func actualizar() -> Bool {
        super.actualizar()
        m_animacion.actualizar()
        return false
    }

    override func dibujar(_ g: Video) {
        guard let mapa = Objeto.mapa else { return }
        m_animacion.dibujar(g, m_x + mapa.tileAncho / 2, m_y + mapa.tileAlto / 2, 0)
    }

    // MARK: - Métodos propios

    func setearPosicion(_ i: Int, _ j: Int) {
        m_posEnTileFisico = (i, j)
        let p = transformarIJEnXY(i, j)
        m_posEnMundoPlano = p
        m_posEnMundoPlano.x -= m_animacion.offsets.x
        m_posEnMundoPlano.y -= m_animacion.offsets.y
        actualizarPosicionXY()
    }
}
